src/espnow.rs

```diff
@@ -0,0 +1,183 @@
+use log::info;
+
+use esp_idf_hal::mutex::Mutex;
+use esp_idf_sys::*;
+
+static RECV_CALLBACK: Mutex<Option<Box<dyn FnMut(&[u8], &[u8]) + Send>>> = Mutex::new(None);
+static SEND_CALLBACK: Mutex<Option<Box<dyn FnMut(&[u8], SendStatus) + Send>>> = Mutex::new(None);
+
+pub static BROADCAST: [u8; 6] = [0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF];
+
+#[derive(Debug)]
+pub enum SendStatus {
+    SUCCESS = 0,
+    FAIL,
+}
+
+impl From<u32> for SendStatus {
+    fn from(val: u32) -> Self {
+        match val {
+            0 => SendStatus::SUCCESS,
+            1 => SendStatus::FAIL,
+            _ => panic!("Wrong status code"),
+        }
+    }
+}
+
+pub type PeerInfo = esp_now_peer_info_t;
+
+#[derive(Debug)]
+struct PrivateData;
+
+#[derive(Debug)]
+pub struct EspNowClient(PrivateData);
+
+impl EspNowClient {
+    pub fn new() -> Result<Self, EspError> {
+        // disable modem sleep, otherwise messages queue up and we're not able
+        // to send any esp-now data after a few messages
+        // esp-idf bug report: https://github.com/espressif/esp-idf/issues/7496
+        esp!(unsafe { esp_wifi_set_ps(0) })?;
+
+        info!("Initializing ESP NOW");
+        esp!(unsafe { esp_now_init() })?;
+        Ok(Self(PrivateData))
+    }
+
+    pub fn new_with_callback(
```

Comment:
Is operation without a receiver callback even feasible? If not, perhaps we just need one constructor (this one)?
Also, given that this is a constructor for the receiver callback, it should be named perhaps `new_with_recv_callback`. 

Or - if you decide to also always provide a sender callback, perhaps: `new_with_callbacks`?

Reply:
I think using espnow without callbacks still makes sense - someone might be just interested in broadcasting the data and not necessarily processing any incoming messages. Will update 👍